src/Radarr.Http/Extensions/RequestExtensions.cs

```diff
@@ -95,10 +95,16 @@ public static string GetRemoteIP(this NancyContext context)
             }
 
             var remoteAddress = context.Request.UserHostAddress;
-            IPAddress remoteIP;
+
+            var realIP = IPAddress.TryParse(remoteAddress, out IPAddress remoteIP);
```

Comment:
I'm not sure you need to set realIP here, as you are spitting out remoteIP already from the tryparse. Might be able to simplify a bit?


```
if (IPAddress.TryParse(remoteAddress, out IPAddress remoteIP) && remoteIP.IsIPv4MappedToIPv6)
{                
  remoteIP = remoteIP.MapToIPv4();
}

// Only check if forwarded by a local network reverse proxy                    
if (remoteIP && remoteIP.IsLocalAddress())
```

Or does it make more sense to make this change in isLocalAddress? Is that what is incorrectly marking it local when it's not?

Reply:
What happens if TryParse returns false - we previously had that Boolean inside of the if block. 

We could move it to inside of local address but so we ever want to return an IPv4 mapped as an IPv6? My reasoning was no hence why I was doing it here as opposed to inside of IsLocalAddress

Reply:
If tryparse is false then remoteIP is null, it should be handled same as you current logic I believe since realIP would be null if it failed parse.

sounds good on the other.